modules/ROOT/pages/batch-filters-and-batch-aggregator.adoc

```diff
@@ -190,34 +190,38 @@ Using random access, you can change a record's payload at any given index positi
 
 ==== Considerations for Defining a Block Size
 
-On a traditional on-line processing model, each request is usually mapped to a worker thread. Regardless of the processing type (either synchronous, asynchronous, one-way, request-response or even if the requests are temporarily buffered before being processed), servers usually end up in a 1:1 relationship between a request and a running thread. +
-When it comes to a batch job, all records are first stored in a persistent queue before the Process phase begins, so that the traditional threading model wouldn't apply.
+In a traditional online processing model, each request is usually mapped to a worker thread. Regardless of the processing type (either synchronous, asynchronous, one-way, request-response or even if the requests are temporarily buffered before being processed), servers usually end up in a 1:1 relationship between a request and a running thread.
 
-To improve performance, the runtime queues and schedules batch records in blocks of 100 records. This lessens the amount of I/O requests and improves an operation's load. +
-The default threading profile of the runtime is 16 threads per job. Therefore, in a default configured batch job each of the 16 threads processes a block of 100 records. +
-Each thread iterates through that block processing each record, and then each block is queued back, and the process continues.
+For a Batch Job, all records are first stored in a persistent queue before the Process phase begins, which means that the traditional threading model does not apply.
 
-Consider having 1 million records to place in a queue for a 3 step batch job. At least three million I/O operations occur as the runtime takes and requests each record as they move through the job's phases. +
-Performance requires having enough available memory to process the 16 threads in parallel, which means moving 1600 records from persistent storage into RAM. The larger your records and their quantity, the more available memory you need for batch processing.
+To improve performance, the runtime queues and schedules batch records in blocks of 100 records. This setting lessens the amount of I/O requests and improves an operation's load.
 
-Although, the standard model of 16 threads, with 100 records per batch job works for most use cases, consider three use cases where you might need to increase or decrease the block size:
+Batch jobs use Mule's thread pool.
+Each thread iterates through a block of up to 100 record. Then each block is queued back, and the process continues.
```

Comment:
this is wrong. It's not `up to 100 records`. It's of the exact block size number (unless there're not enough records)